docs/apps/hyperqueue.md

```diff
@@ -0,0 +1,229 @@
+    echo "Checking if workers have started"
+    if [[ $num_up -eq $SLURM_NNODES ]];then
+        echo "Workers started"
+        break
+    fi
+    echo "$num_up/$SLURM_NNODES workers have started"
+    sleep 1
+    num_up=$(hq worker list | grep RUNNING | wc -l)
+
+done
+```
+
+### Submitting jobs
+
+!!!info Output
+    By default hq creates one folder for each job where output is redirected
+    You can use the --log,--stdout,--stderr flags to change this behavior.
+    Note that it's not possible to direct output from multiple jobs into the same file
+    As each submission will clear the file
+
+See `hq submit --help` for full list of options.
+
+`hq submit <hq_submit args> --cpus <n> <COMMAND/executable> <args to program>`
+
+This is a non-blocking command in the same way as `sbatch`. 
+
+Hyperqueue is not limited to running a single execution per submission. 
+Using the `--array 1-N` flag we can start a program N times similar to how slurm array jobs work.
+
+`hq submit --array 1-10 --cpus <n> <COMMAND>`. `<COMMAND>` Then has access to the environment variable `HQ_TASK_ID`
+which is used to enumerate all the tasks. 
+
+!!!info sbatch-hq
+    For very simple submissions where you only want to run each line within a file with identical resources
+    you can just use the csc utility tool `sbatch-hq`, and not have to care about hyperqueue. `module load sbatch-hq` 
+
+When we have submitted everything we want, we need to wait for the jobs to finish, this can be done e.g with:
+
+```
+echo "WAITING FOR JOBS TO FINISH"
+sleep 30
+NOT_DONE=$(hq job list --all | grep "RUNNING\|PENDING")
+while true; do
+    echo "WAITING FOR JOBS TO FINISH "
+    if [[ -z "$NOT_DONE" ]];then
+        break
+    fi
+    sleep 30
+    NOT_DONE=$(hq job list --all | grep "RUNNING\|PENDING")
+done
+```
+
+Once we are done running all of our jobs, we shutdown the workers and server so that we don't get a false
+error from slurm when the job ends
+
+```bash
+hq worker stop all
+hq server stop
+```
+
+### Full example
+
+```
+#SBATCH --partition=medium
+#SBATCH --account=<project>
+#SBATCH --nodes=4
+#SBATCH --cpus-per-task=128
+#SBATCH --ntasks-per-node=1
+#SBATCH --time=01:00:00
+
+export SLURM_EXACT=1
+
+# Load the required modules
+module load hyperqueue
+
+# Set the directory which hyperqueue will use 
+export HQ_SERVER_DIR=$PWD/hq-server-$SLURM_JOB_ID
+mkdir -p $HQ_SERVER_DIR
+
+echo "STARTING HQ SERVER, log in $HQ_SERVER_DIR/HQ.log"
+echo "===================="
+hq server start &>> $HQ_SERVER_DIR/HQ.log &
+echo "STARTING HQ WORKERS ON $SLURM_NNODES nodes"
+echo "===================="
+srun --cpu-bind=none --mpi=none hq worker start --cpus=$SLURM_CPUS_PER_TASK &>> $HQ_SERVER_DIR/HQ.log &
+
+num_up=$(hq worker list 2>/dev/null | grep RUNNING | wc -l)
+while true; do
+    echo "WAITING FOR WORKERS TO START ( $num_up / $SLURM_NNODES )"
+    if [[ $num_up -eq $SLURM_NNODES ]];then
+        break
+    fi
+    sleep 1
+    num_up=$(hq worker list | grep RUNNING | wc -l)
```

Comment:
`num_up=$(hq worker list 2>/dev/null | grep -c RUNNING)

same as above, `grep -c` is enough an cleaner I think.

But also remove stderr here as wall (since you're doing it already above)

It would be neat to do it at the beginning of the loop (the assignment) to avoid repeating yourself.